File: frontend/src/api/logistics.ts
import { api } from './client';
import type { RerouteRequest, RerouteResponse } from '@/types/reroute';

export const rerouteTrucks = (req: RerouteRequest) =>
  api.post<RerouteResponse>('/logistics/reroute', req).then(r => r.data);

File: backend/app/routers/logistics.py
from fastapi import APIRouter

from app.schemas.logistics import RerouteRequest, RerouteResponse
from app.services.routing import reroute

router = APIRouter(prefix="/api/logistics", tags=["logistics"])


@router.post("/reroute", response_model=RerouteResponse)
async def reroute_trucks(req: RerouteRequest) -> RerouteResponse:
    return reroute(req)
